### DISCUSSION TITLE
Unicode error when using custom checkout fields

### DISCUSSION THREAD

**Discussion by gustavoconter** [woocommerce/woocommerce#46557] [WooCommerce, Blocks, and Site Editing]
Hey! I am working on a plugin that adds support for brazilian fields for all brazilian e-commerces that want to migrate to the blocks based checkout. However, I've noticed that when using the `/` in the custom field id. Inputs with accented characters like: `São Paulo` will get saved on the database as `Su00e3o Paulo`. This would be correct if it wasn't for the missing backslash that makes impossible do decode this string back to the normal form.

```php
__experimental_woocommerce_blocks_register_checkout_field(
            array(
                'id'            => 'wcbrf/neighborhoodf',
                'label'         => 'Bairro',
                'location'      => 'address',
                'required'      => true,
                'index' => 30,
            ),
        );
```

Doing some tests, I've discovered that this bug is probably related to the ID. In the moment I added a backslash, the fields began to save correctly in the database. However, using a backslash introduced a new bug, now the fields are not being populated when the checkout is loaded.

```php
__experimental_woocommerce_blocks_register_checkout_field(
            array(
                'id'            => 'wcbrf\/neighborhoodf',
                'label'         => 'Bairro',
                'location'      => 'address',
                'required'      => true,
                'index' => 30,
            ),
        );
```

Any idea on why this might be happening? Is this ID convention the final convention? Perhaps we could migrate to a simple hyfen or underscode.

Any help on this issue would be much appreciated.

**Comment by senadir** [comment]
This seems more like a bug than a discussion, I will convert it to an issue.

**Comment by nielslange** [comment]
Closing this, as it had been converted to the issue #46584 and fixed by the PR #46762.
